src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/RISCV64Context.cs

```diff
@@ -0,0 +1,218 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Diagnostics.DataContractReader.Contracts.StackWalkHelpers.RISCV64;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts.StackWalkHelpers;
+
+/// <summary>
+/// RISC-V 64-bit specific thread context.
+/// </summary>
+[StructLayout(LayoutKind.Explicit, Pack = 1)]
+internal struct RISCV64Context : IPlatformContext
+{
+    [Flags]
+    public enum ContextFlagsValues : uint
+    {
+        CONTEXT_RISCV64 = 0x00800000,
+        CONTEXT_CONTROL = CONTEXT_RISCV64 | 0x1,
+        CONTEXT_INTEGER = CONTEXT_RISCV64 | 0x2,
+        CONTEXT_FLOATING_POINT = CONTEXT_RISCV64 | 0x4,
+        CONTEXT_DEBUG_REGISTERS = CONTEXT_RISCV64 | 0x8,
+        CONTEXT_FULL = CONTEXT_CONTROL | CONTEXT_INTEGER | CONTEXT_FLOATING_POINT,
+        CONTEXT_ALL = CONTEXT_CONTROL | CONTEXT_INTEGER | CONTEXT_FLOATING_POINT | CONTEXT_DEBUG_REGISTERS,
+
+        //
+        // This flag is set by the unwinder if it has unwound to a call
+        // site, and cleared whenever it unwinds through a trap frame.
+        // It is used by language-specific exception handlers to help
+        // differentiate exception scopes during dispatching.
+        //
+        CONTEXT_UNWOUND_TO_CALL = 0x20000000,
+        CONTEXT_AREA_MASK = 0xFFFF,
+    }
+
+    public readonly uint Size => 0x300; // Approximate size, may need adjustment
```

Comment:
It looks the diagnostics repo uses a size of `0x220`. Did using this size cause failures?

https://github.com/microsoft/clrmd/blob/d724947392626b66e39b525998a8817447d50380/src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs#L20

Reply:
Ah, not I haven't tested it end to end:
```suggestion
    public readonly uint Size => 0x220;
```